File: LeetCode root folder/Count the Number of Houses at a Certain Distance I.cpp
using namespace std;
class Solution {
  public:
      vector<int> countOfPairs(int n, int x, int y) {
        unordered_map<int, int> map;
        for (int i = 1; i <= n; i++) {
          for (int j = i + 1; j <= n; j++) {
            int minDist = min({
                j - i,
                1 + abs(i - x) + abs(j - y),
                1 + abs(j - x) + abs(i - y)
          });
            map[minDist] += 2;
          }
        }
        vector<int> ans(n, 0);
        for (int i = 0; i < n; i++) {
          ans[i] = (map[i+1]);
        }
        return ans;
      }
  };

File: LeetCode root folder/Count the Number of Houses at a Certain Distance I.py
# greedy solution:150
# ms
# Beats
# 76.64%
class Solution:
    def countOfPairs(self, n: int, x: int, y: int) -> List[int]:
        hashMap = defaultdict(int)
        for i in range(1, n+1):
            for j in range(i+1, n+1):
                # i - j
                # "(y - x)" 1 + abs(i-x) + abs(j-y)
                # "(y - x)" 1 + abs(j-x) + abs(i-y)
                minDist = min(
                    j - i,
                    1 + abs(i-x) + abs(j-y),
                    1 + abs(j-x) + abs(i-y)
                )
                hashMap[minDist] += 2

        ans = []
        for i in range(1, n+1):
            ans.append(hashMap[i])
        return ans
